contracts/farm/FeeExtractAndDistribute_BSC.sol

```diff
@@ -55,95 +47,57 @@ contract FeeExtractAndDistribute_BSC is Upgradeable {
         uint256 dstAmount
     );
 
-    event AssetBurn(
-        address indexed sender,
-        address indexed asset,
-        uint256 amount
-    );
-
-    modifier onlyEOA() {
-        require(msg.sender == tx.origin, "unauthorized");
-        _;
-    }
-
-    modifier checkPause() {
-        require(!isPaused || msg.sender == owner(), "paused");
-        _;
-    }
-
-    function sweepFees() public // sweepFeesByAsset() does checkPause
-    {
-        sweepFeesByAsset(currentFeeTokens);
+    function sweepFees() public pausable {
+        _extractAndDistribute(currentFeeTokens);
     }
 
-    function sweepFeesByAsset(address[] memory assets)
-        public
-        checkPause
-        onlyEOA
-    {
+    function sweepFees(address[] memory assets) public pausable {
         _extractAndDistribute(assets);
     }
 
     function _extractAndDistribute(address[] memory assets) internal {
-        uint256[] memory amounts = bZx.withdrawFees(
+        uint256[] memory amounts = BZX.withdrawFees(
             assets,
             address(this),
             IBZx.FeeClaimType.All
         );
 
         for (uint256 i = 0; i < assets.length; i++) {
-            require(assets[i] != BGOV, "asset not supported");
-            exportedFees[assets[i]] = exportedFees[assets[i]].add(amounts[i]);
+            exportedFees[assets[i]] += amounts[i];
         }
 
-        uint256 bnbOutput = exportedFees[BNB];
-        exportedFees[BNB] = 0;
+        uint256 usdcOutput = exportedFees[USDC];
+        exportedFees[USDC] = 0;
 
         address asset;
         uint256 amount;
         for (uint256 i = 0; i < assets.length; i++) {
             asset = assets[i];
-            if (asset == BGOV || asset == BZRX || asset == BNB) {
-                continue;
-            }
+            if (asset == USDC) continue; //USDC already accounted for
```

Comment:
there is no USDC on BSC

Reply:
I know but for the bridging of assets, it will use USDC to make things standardized. I can remove the check here though